test/plugins/test_implementations.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2017-2020 Palantir Technologies, Inc.
+# Copyright 2021- Python Language Server Contributors.
+
+from collections.abc import Iterable
+from importlib.resources import as_file, files
+from pathlib import Path
+
+import pytest
+from rope.base.exceptions import BadIdentifierError
+
+from pylsp import uris
+from pylsp.config.config import Config
+from pylsp.plugins.rope_implementation import pylsp_implementations
+from pylsp.workspace import Workspace
+
+
+# We use a real file because the part of Rope that this feature uses
+# (`rope.findit.find_implementations`) *always* loads files from the
+# filesystem, in contrast to e.g. `code_assist` which takes a `source` argument
+# that can be more easily faked.
+# An alternative to using real files would be `unittest.mock.patch`, but that
+# ends up being more trouble than it's worth...
+@pytest.fixture
+def examples_dir_path() -> Iterable[Path]:
+    with as_file(files("test.data.implementations_examples")) as path:
```

Comment:
Regarding the failing tests, this only works in Python 3.12+. I'll try to see how it can be made to work in older versions.